generator/src/test/java/com/scottlogic/deg/generator/restrictions/MustContainRestrictionTests.java

```diff
@@ -0,0 +1,77 @@
+package com.scottlogic.deg.generator.restrictions;
```

Comment:
This test would be easier to read, if we didn't create a mock field spec. 

could alternatively put 

        FieldSpec nullRestriction = FieldSpec.Empty.withNullRestrictions(new 
        NullRestrictions(Nullness.MUST_NOT_BE_NULL), null);
    FieldSpec typeRestrictions = FieldSpec.Empty.withTypeRestrictions(DataTypeRestrictions.ALL_TYPES_PERMITTED, null);

and directly create mustContainRestrictions in the test

            MustContainRestriction a = new MustContainRestriction(Stream.of(nullRestriction, typeRestrictions).collect(Collectors.toSet()));

would allow fieldSpec to remain private

Reply:
In part the reason was to separate the concerns of the test, but it could tidy-up the tests a little.

Reply:
Changed, and `FieldSpec` constructor returned to being private.